core-to-plc/src/Language/Plutus/CoreToPLC/Plugin.hs

```diff
@@ -169,7 +170,9 @@ convertExpr opts origE tpe = do
                   annotated <- convertErrors (NoContext . PLCError) $ PLC.annotateTerm converted
                   void $ convertErrors (NoContext . PLCError) $ PLC.typecheckTerm 1000 annotated
               pure converted
-    case runExcept $ runQuoteT $ evalStateT (runReaderT result (flags, primTerms, primTys, initialScopeStack)) Map.empty of
+        context = (ConversionOptions { coCheckValueRestriction=poDoTypecheck opts }, flags, primTerms, primTys, initialScopeStack)
+        initialState = ConvertingState Map.empty
+    case runExcept $ runQuoteT $ evalStateT (runReaderT result context) initialState of
```

Comment:
I think it's best to have a separate function for this (`runConverting` or something).

Reply:
Sure, but I'll do it in a follow up.